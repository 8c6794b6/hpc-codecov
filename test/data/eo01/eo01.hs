{-# LANGUAGE GeneralizedNewtypeDeriving #-}
module Main where

data R = R
  { field1 :: Bool
  , field2 :: Bool
  }

pm01 :: Maybe Int -> Int
pm01 (Just n) =
  n + 1
pm01 Nothing =
  12345

br01 :: Int -> Int -> Int
br01 a b
  | a < b = b
  | a == 0 = b
  | b < 0 = a + b
  | otherwise = 12345

newtype A f x = A {unA :: f x}
  deriving (Functor, Applicative, Monad)

newtype B f x = B {unB :: f x}

instance Functor f => Functor (B f) where
  fmap f (B fx) = B (fmap f fx)

br02 :: Int -> Int -> Int -> Int
br02 a b c
  | a < b = b
  | a == 0 = b
  | b < c = a + b
  | otherwise = 12345

main :: IO ()
main = do
  case R True False of
    R t f -> print (t && f)
  print $ pm01 Nothing
  print $ br01 2 1
  print $ br02 0 (-1) 0
