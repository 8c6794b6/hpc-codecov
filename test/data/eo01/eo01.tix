Tix [ TixModule "Main" 1101637578 105 [1,1,0,1,1,0,1,1,1,0,1,1,0,0,0,0,0,0,0,0,0,0,0,0,1,1,1,0,1,1,0,1,1,0,1,1,0,1,1,0,1,1,0,0,0,1,0,1,1,1,0,0,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]]
