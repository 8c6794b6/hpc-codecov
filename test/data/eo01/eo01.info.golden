TN:
SF:test/data/eo01/eo01.hs
FNF:0
FNH:0
BRF:0
BRH:0
DA:5,0
DA:6,0
DA:11,0
DA:13,1
DA:17,1
DA:18,1
DA:19,1
DA:20,1
DA:22,0
DA:25,0
DA:28,0
DA:32,1
DA:33,1
DA:34,0
DA:35,0
DA:38,1
DA:39,1
DA:40,1
DA:41,1
DA:42,1
DA:43,1
LF:21
LH:13
end_of_record
